### PR TITLE
fix: clean up lint warnings

### LINKED ISSUE #2605: fix lint warnings
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

We have a lot of lint warnings, let's clean it up a bit
```
The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 38.
Checked 1145 files in 301ms. Fixed 1 file.
Found 58 warnings.
```

### Steps to Reproduce

run `pnpm install && pnpm build && pnpm fmt`

### Expected behavior

fewer lint warnings

### Other information

_No response_

### Screenshots

_No response_

### Version info

```markdown
- OS:
- Node:
- npm:
```

### PR DESCRIPTION
**Pull request by harshsbhat**
## What does this PR do?

Fixes #2605

Since the Lint rules were made strict, many new warnings appeared.  This PR cleans up those warnings

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Run `pnpm fmt`
- Note that there are no more warnings
### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Release Notes

- **New Features**
  - Enhanced error handling for image uploads and curl command parsing.
  - Improved responsiveness of components to prop changes with updated dependency management.

- **Bug Fixes**
  - Streamlined logic for form submissions and state management in various components.

- **Refactor**
  - Optimized useEffect and useMemo hooks across multiple components for better performance and clarity.

- **Chores**
  - Cosmetic changes, such as renaming variables for improved code readability.

### CODE REVIEW

**Comment by harshsbhat** [comment]
I am not 1000% sure about one or two of them. I apologize in advance if something is messed up.

File: apps/dashboard/components/dashboard/visible-button.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function VisibleButton({
     setTimeout(() => {
       setIsVisible(false);
     }, 10000);
-  }, [isVisible]);
+  }, [setIsVisible]);
```
[the remark this answers is not in the document]
Comment (harshsbhat) [reply apps/dashboard/components/dashboard/visible-button.tsx:21]: This was giving Lint warning

File: apps/www/components/copy-button.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function CopyButton({ value, className, src, children, ...props }: CopyBu
     setTimeout(() => {
       setHasCopied(false);
     }, 2000);
-  }, [hasCopied]);
+  }, []);
```
[the remark this answers is not in the document]
Comment (unrenamed) [reply apps/www/components/copy-button.tsx:23]: the bot forgot about the `if` statement to check if `hasCopied` is not false — if it's, the effect makes no sense, does it?

File: apps/www/components/particles.tsx [hunk 3]
```diff
@@ -61,11 +60,11 @@ export const Particles: React.FC<ParticlesProps> = ({
 
   useEffect(() => {
     onMouseMove();
-  }, [mousePosition.x, mousePosition.y]);
+  }, []);
```
[the remark this answers is not in the document]
Comment (unrenamed) [reply apps/www/components/particles.tsx:63]: the bot is right. the same logic as in [apps/www/components/shiny-card.tsx](https://github.com/unkeyed/unkey/pull/2648/files#diff-f49b38a359ae538213770ba2ebae7e892ad175e347c471a5059a351d63a20626)

File: apps/www/components/particles.tsx [hunk 4]
```diff
@@ -61,11 +60,11 @@ export const Particles: React.FC<ParticlesProps> = ({
...
 
   useEffect(() => {
     initCanvas();
-  }, [refresh]);
+  }, []);
```
Comment (unrenamed) [inline apps/www/components/particles.tsx:67]: I'd suggest checking how it affected the component behaviour. smth tells me this effect re-initiates canvas on refresh, hence `refresh` might be necessary here. the same logic as in [apps/www/components/shiny-card.tsx](https://github.com/unkeyed/unkey/pull/2648/files#diff-f49b38a359ae538213770ba2ebae7e892ad175e347c471a5059a351d63a20626)

File: apps/www/components/shiny-card.tsx [hunk 5]
```diff
@@ -10,11 +10,7 @@ type ShinyCardGroupProps = {
   refresh?: boolean;
 };
 
-export const ShinyCardGroup: React.FC<ShinyCardGroupProps> = ({
-  children,
-  className = "",
-  refresh = false,
-}) => {
+export const ShinyCardGroup: React.FC<ShinyCardGroupProps> = ({ children, className = "" }) => {
```
[the remark this answers is not in the document]
Comment (unrenamed) [reply apps/www/components/shiny-card.tsx:13]: I suggest you don't rush with this one. see the comment below. the prop may be still required

File: apps/dashboard/app/(app)/settings/root-keys/new/client.tsx [hunk 6]
```diff
@@ -113,7 +113,7 @@ export const Client: React.FC<Props> = ({ apis }) => {
     // for large data sets.
 
     setCardStatesMap(initialCardStates);
-  }, []); // Execute ones on the first load
+  }, [apis, selectedPermissions]); // Execute ones on the first load
```
Comment (unrenamed) [inline apps/dashboard/app/(app)/settings/root-keys/new/client.tsx:116]: the comment explicitly states that the effect should only run once. Adding dependencies breaks this intended behavior.

File: apps/www/components/ui/meteorLines.tsx [hunk 7]
```diff
@@ -78,7 +78,7 @@ const MeteorLinesAngular = ({
       animationDuration: speed ? `${speed}s` : `${Math.floor(Math.random() * 10 + 2)}s`,
     }));
     setMeteorStyles(styles);
-  }, [number, windowWidth]);
```
Comment (unrenamed) [inline apps/www/components/ui/meteorLines.tsx:81]: the new change doesn’t seem to break existing functionality, but it looks like it wasn’t working as expected before. `windowWidth` isn’t updated on window resize—it’s set once by the effect and never changes, leading to issues like the meteor lines staying on the left when resizing from tablet to full HD (see screenshot).

image

cc @MichaelUnkey, Git blame shows you last modified this component. I think the core team should decide whether to fix this. It does not seem critical to me.

File: apps/www/components/ui/hover-border-gradient.tsx [hunk 8]
```diff
@@ -55,7 +55,7 @@ export function HoverBorderGradient({
       }, duration * 1000);
       return () => clearInterval(interval);
     }
-  }, [hovered]);
+  }, [hovered, rotateDirection, duration]);
```
Comment (unrenamed) [inline apps/www/components/ui/hover-border-gradient.tsx:58]: the `HoverBorderGradient` component doesn't seem to be used anymore. Should we remove it entirely? cc @chronark

File: apps/dashboard/app/(app)/settings/root-keys/[keyId]/permissions/add-permission-for-api.tsx [hunk 9]
```diff
@@ -40,7 +40,7 @@ export function DialogAddPermissionsForAPI(
   const [selectedApiId, setSelectedApiId] = useState<string>("");
   const selectedApi = useMemo(
     () => props.apis.find((api) => api.id === selectedApiId),
-    [selectedApiId],
+    [selectedApiId, props.apis],
```
Comment (unrenamed) [inline apps/dashboard/app/(app)/settings/root-keys/[keyId]/permissions/add-permission-for-api.tsx:43]: frequent changes to `props.apis` can hurt performance, as it's an array of objects and shallow comparison is used for dependencies. Ensure `props.apis` doesn't change often.

Comment (harshsbhat) [reply apps/dashboard/app/(app)/settings/root-keys/[keyId]/permissions/add-permission-for-api.tsx:43]: @unrenamed you are right. See my comment. It was pretty weird to see these things show a lint warning. I will refactor this entirely. Archiving for now

**Comment by unrenamed** [comment]
Lint warnings aren’t always style issues; they can indicate that components (in the React case) are poorly written and need refactoring to align with React design principles. Some of your changes actually confirmed that so it's great that you started addressing the lint warnings.

> @unrenamed you are right. See my comment. It was pretty weird to see these things show a lint warning. I will refactor this entirely. Archiving for now

Okay, but leaving "broken windows" is not ideal. This can be tackled in stages: either within this PR (just not addressing all points at once) or in future PRs where you're fixing other bugs.

**Comment by harshsbhat** [comment]
@unrenamed  I think I messed up pretty badly here. Closing this for now. Feel free to take over or create a new PR. Sorry once again for a bad PR.  

cc: @chronark

**Pull request closed without merging**